numpy-stubs/__init__.pyi

```diff
@@ -898,6 +913,23 @@ def partition(
     kind: _PartitionKind = ...,
     order: Union[None, str, Sequence[str]] = ...,
 ) -> ndarray: ...
+@overload
```

Comment:
The behaviour of ``np.argpartition()`` here is a bit odd.
If an ``np.ndarray`` is passed an ``np.ndarray`` is returned and if a ``np.generic`` is passed a ``np.generic`` is returned, so far so good.

However, if a builtin scalar is passed it also returns a ``np.ndarray``, rather than a ``np.generic``.

```python
>>> import numpy as np

>>> type(np.argpartition(0, 0))
numpy.ndarray

>>> type(np.argpartition(np.int64(0), 0))
numpy.int64
```